### PR TITLE
Test properties and custom events in vue js components with jest: update

### PR DESCRIPTION
**Pull request by lex111**
- [x] Sync with master
- [x] Upgrade to latest version of vue-test-utils
- [x] Make all tests green :smile: 
- [x] Fix small typos
- [x] Update corresponding post (see https://github.com/alexjoverm/blog/pull/4)

cc @alexjoverm

### CODE REVIEW

File: test/Message.test.js [hunk 1]
```diff
@@ -47,21 +46,25 @@ describe('Message.test.js', () => {
...
     it('triggers a message-clicked event when a handleClick method is called', () => {
       const stub = jest.fn()
       cmp.vm.$on('message-clicked', stub)
-
       cmp.vm.handleClick()
+
       expect(stub).toBeCalledWith('Cat')
     })
+
+    it('calls handleMessageClick when @message-click happens', () => {
```
Comment (lex111) [inline test/Message.test.js:64]: From [post](https://github.com/alexjoverm/blog/blob/7bdbeefb130fc708debbb4849124e86966c3daa8/source/_posts/Test-Properties-and-Custom-Events-in-Vue-js-Components-with-Jest.md#wrapping-up):

> I'll leave up to you to test what `handleMessageClicked` does ;).

This was meant? I added to check this offer, because didn't quite understand it. About him is not described in the post, although in general you can mention, but if there's anything I can remove from here.

cc @alexjoverm

Comment (alexjoverm) [reply test/Message.test.js:64]: @lex111 don't quite understand what you mean. Do you mean that sentence is not understandable? If so, maybe we can rephrase it to something like:

> This test checks that the `message-clicked` event is emitted, mocking the `handleMessageClick` method, but we still have to test what that method does in a separate test. I'll leave that to you as an exercise ;).

Is it more clear like that?

Comment (lex111) [reply test/Message.test.js:64]: Yes, I see, thanks, but this test I've marked is not the solution for this exercise?

See also https://github.com/alexjoverm/vue-testing-series/pull/18/files#r210841763

File: test/MessageList.test.js [hunk 2]
```diff
@@ -56,26 +56,23 @@ describe('MessageList.test.js', () => {
...
-    // const stub = jest.fn()
-    // cmp.setMethods({ handleMessageClick: stub })
+    const stub = jest.fn()
+    cmp.setMethods({ handleMessageClick: stub })
+    const el = cmp.find('.message').vm.$emit('message-clicked', 'Cat')
 
-    const el = cmp.find(Message).vm.$emit('message-clicked', 'cat')
-
-    expect(cmp.vm.handleMessageClick).toBeCalledWith('cat')
+    expect(stub).toBeCalledWith('Cat')
```
Comment (lex111) [inline test/MessageList.test.js:76]: More precisely, it's already done?

File: test/Message.test.js (hunk 1 above)
Comment (lex111) [reply test/Message.test.js:64]: In this test, I can determine from the parent component that the event was emitted using `emitted`.

And you do not need to mock anything.

Comment (alexjoverm) [reply test/Message.test.js:64]: Not really...

There are 2 tests you can do for that:

 1 - Event emitted: in this test you test that the message-click event is handled. We don't care about what the handler implementation is, thus we use a mock.
 2 - Handler test: in this test, we test what the parent handler does. We don't mock it and we don't need to call any event to trigger it, just call the method directly.

The exercise suggested is for the second case.

Comment (lex111) [reply test/Message.test.js:64]: I'm actually not really the point of this testing, okay, what do I do with this PR? Delete this test? It simply shows the additional possibility with the help of `emitted`.

Comment (alexjoverm) [reply test/Message.test.js:64]: I'd suggest to keep it as close to the original book as possible. It's a good practise for the test to be unitary and atomic, that's why is usually better to keep them simple and separated.

Additionally, you can also remove the suggestion of the exercise if you don't like it :).

Comment (alexjoverm) [reply test/Message.test.js:64]: The tests you wrote look good by the way, you can keep them as that :)

Comment (lex111) [reply test/Message.test.js:64]: Then you can accept this PR?

Comment (alexjoverm) [reply test/Message.test.js:64]: Oh, sure :)

Comment (lex111) [reply test/Message.test.js:64]: Don't forget about this https://github.com/alexjoverm/blog/pull/4
